central/debug/service/service.go

```diff
@@ -502,6 +508,16 @@ func (s *serviceImpl) writeZippedDebugDump(ctx context.Context, w http.ResponseW
 
 	zipWriter := zip.NewWriter(w)
 
+	s.writeZippedDebugDumpWithTimeout(ctx, w, opts, zipWriter)
```

Comment:
(pending on the comment I added about generally handling timeouts, i.e. setting timeout specifically vs request cancellations):

If you hit a timeout, I appreciate the warning you have added to the diagnostic bundle, but shouldn't we strive to responde with an actual error in this case?
I know we currently fail-open for other errors, but in case of a timeout, the diagnostic bundle essentially becomes quite useless, so responding with an explicit error + instruction to increase timeout personally sounds like a better behavior of the API to me.